addons/binding/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/model/HttpChannelConfig.java

```diff
@@ -0,0 +1,139 @@
+     * Returnes a parsed state request configuration, if one was specified.
+     *
+     * @param bundleContext the OSGi bundle context
+     * @return an optional {@link StateRequest}
+     * @throws InvalidConfigurationException if any part of the configuration is not valid
+     */
+    public Optional<StateRequest> getStateRequest(final BundleContext bundleContext) {
+        return Optional.ofNullable(this.stateUrl).map(stateUrl -> {
+            final Optional<Transform> stateResponseTransform = parseTransform(
+                    bundleContext, "stateResponseTransform", this.stateResponseTransform);
+            try {
+                return new StateRequest(
+                        new URL(stateUrl),
+                        Optional.ofNullable(stateUsername),
+                        Optional.ofNullable(statePassword),
+                        Duration.ofMillis(this.stateConnectTimeout),
+                        Duration.ofMillis(this.stateRequestTimeout),
+                        Duration.ofMillis(this.stateRefreshInterval),
+                        stateResponseTransform
+                );
+            } catch (final MalformedURLException e) {
+                throw new InvalidConfigurationException("Invalid stateUrl: " + e.getMessage());
+            }
+        });
+    }
+
+    /**
+     * Returnes a parsed command request configuration, if one was specified.
+     *
+     * @param bundleContext the OSGi bundle context
+     * @return an optional {@link CommandRequest}
+     * @throws InvalidConfigurationException if any part of the configuration is not valid
+     */
+    public Optional<CommandRequest> getCommandRequest(final BundleContext bundleContext) {
+        return Optional.ofNullable(this.commandUrl).map(commandUrl -> {
+            final Optional<Transform> commandRequestTransform = parseTransform(
+                    bundleContext, "commandRequestTransform", this.commandRequestTransform);
+            final Optional<Transform> commandResponseTransform = parseTransform(
+                    bundleContext, "commandResponseTransform", this.commandResponseTransform);
+            try {
+                return new CommandRequest(
+                        this.commandMethod,
+                        new URL(commandUrl),
+                        Optional.ofNullable(commandUsername),
+                        Optional.ofNullable(commandPassword),
+                        Duration.ofMillis(this.commandConnectTimeout),
+                        Duration.ofMillis(this.commandRequestTimeout),
+                        this.commandContentType,
+                        commandRequestTransform,
+                        commandResponseTransform
+                );
+            } catch (final MalformedURLException e) {
+                throw new InvalidConfigurationException("Invalid commandUrl: " + e.getMessage());
+            }
+        });
+    }
+
+    private Optional<Transform> parseTransform(final BundleContext bundleContext,
```

Comment:
I haven't found any problems anymore, but here is one big performance tip: It absolutely makes sense to cache the `TransformationService`. At the moment you are resolving this service for every http call (that uses transforms).

Imagine this: A user polls a rest interface every second with 10 different endpoint calls (doesn't sound unusual). So you are resolving this service 10 times a second.

And resolving a service is slllloooowwww. You are calling out to the OSGI framework and OSGI does a lot of string comparisons to finally find the service and hand it over, all thread safe and expensive of course.

Ideally just resolve it once per handler.